### PR TITLE
PIC-3295 return dynamic counts by hearing outcome state

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/CaseWorkflowController.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class CaseWorkflowController(val caseWorkflowService: CaseWorkflowService, val a
     @Operation(description = "Fetch hearing outcomes")
     @GetMapping(value = ["/courts/{courtCode}/hearing-outcomes"], produces = [APPLICATION_JSON_VALUE])
     fun fetchHearingOutcomes(@PathVariable("courtCode") courtCode: String, @Valid hearingOutcomeSearchRequest: HearingOutcomeSearchRequest): HearingOutcomeCaseList {
-        val cases = caseWorkflowService.fetchHearingOutcomes(courtCode, hearingOutcomeSearchRequest)
-        return HearingOutcomeCaseList(cases)
+        val cases = caseWorkflowService.getOutcomeCountsByState(courtCode, hearingOutcomeSearchRequest)
```
Comment (vb-moj) [inline src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/CaseWorkflowController.kt:36]: The name is misleading. getOutcomeByState() but the state is not part of it! what is wrong with the previous name?

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt [hunk 2]
```diff
@@ -76,4 +76,21 @@ class HearingOutcomeRepositoryCustom(private val entityManager: EntityManager) {
 
         return jpaQuery.resultList.map { it as Array<Any> }.map { Pair(it[0] as HearingEntity, it[1] as LocalDate) };
     }
+
+    fun getDynamicOutcomeCountsByState(courtCode: String): Map<String, Int> {
```
Comment (vb-moj) [inline src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:80]: same here, I can see the state is involved in the query but struggling to understand the relevance of it in the name. please clarify.

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/CaseWorkflowController.kt (hunk 1 above)
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/CaseWorkflowController.kt:36]: Sorry, that was not supposed to be renamed. Let me fix

Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/CaseWorkflowController.kt]: fixed

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/model/HearingOutcomeCountByState.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.probation.courtcaseservice.controller.model
+
+import io.swagger.v3.oas.annotations.media.Schema
+
+@Schema(description = "Hearing outcome counts by workflow state.")
+data class HearingOutcomeCountByState(
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/controller/model/HearingOutcomeCountByState.kt:6]: That's counts by state

`    val toResultCount: Int,
    val inProgressCount: Int,
    val resultedCount: Int,`

**Pull request merged**